metricbeat/module/kafka/partition/partition.go

```diff
@@ -19,75 +21,196 @@ func init() {
 // MetricSet type defines all fields of the partition MetricSet
 type MetricSet struct {
 	mb.BaseMetricSet
-	client sarama.Client
+
+	broker *sarama.Broker
+	cfg    *sarama.Config
+	id     int32
 }
 
-// New creates a new instance of the partition MetricSet
+var noID int32 = -1
+
+var errFailQueryOffset = errors.New("Failed to query offset for")
+
+// New create a new instance of the partition MetricSet
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
-	logp.Warn("EXPERIMENTAL: The %v %v metricset is experimental", base.Module().Name(), base.Name())
+	config := struct{}{}
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, err
+	}
+
+	cfg := sarama.NewConfig()
+	cfg.Net.DialTimeout = base.Module().Config().Timeout
+	cfg.Net.ReadTimeout = base.Module().Config().Timeout
+	cfg.ClientID = "metricbeat"
+
+	broker := sarama.NewBroker(base.Host())
+	return &MetricSet{
+		BaseMetricSet: base,
+		broker:        broker,
+		cfg:           cfg,
+		id:            noID,
+	}, nil
+}
+
+func (m *MetricSet) connect() (*sarama.Broker, error) {
+	b := m.broker
+	if err := b.Open(m.cfg); err != nil {
+		return nil, err
+	}
```

Comment:
Not sure. But my thinking here is similar to my comment below. It is ok if it fails from time to time, as we call connect() for each Fetch. I prefer that it fails instead of waiting for a long time.